Fix Lev_fiber.waitpid on Windows

mnxn opened the pull request:
Modifications so far:
- Forcing `process_thread` after adding a process to the `active` table
- Always recursively call `run_thread`
- Using `Fdecl.set` to set `tref` (it wasn't being set at all before, causing an exception)

Now it seems that the `ivar` does get filled, but then it runs in an infinite loop.

I think I might be misunderstanding how this function is supposed to work. Is it supposed to be running in a separate thread or something? It doesn't look like it's doing that now.

rgrinberg commented:
> Forcing process_thread after adding a process to the active table

I don't think this change should make a difference. Although this change is harmless.

> I think I might be misunderstanding how this function is supposed to work. Is it supposed to be running in a separate thread or something? It doesn't look like it's doing that now.

Yes indeed. The function should be running in a separate thread. We should wrap this with `Thread.create`.

mnxn commented:
> I don't think this change should make a difference. Although this change is harmless.

I see that now. I dropped that commit.

> The function should be running in a separate thread. We should wrap this with Thread.create.

Yes, that's it! I was overlooking this before. I added the thread creation and everything (including formatting in the language server) seems to work now!

However, I did use the private `Lev_fiber.Thread.spawn_thread` function to create the thread. I wasn't sure if you meant `Thread.create` from the standard library or from `Lev_fiber`. The standard library `Thread` module is shadowed and I can't call it directly. I'm not sure how to use `Lev_fiber.Thread.create` in this context if that's what you meant instead.
